global/functions/spawning/fn_spawning_find_valid_position_tracer.sqf

```diff
@@ -77,24 +131,28 @@ while {true} do {
 
 	//If we find a unit, we exit and set the last valid position + which target stopped us.
 	if (!_positionIsValid) exitWith {
-		if (_tracerPosition isEqualTo _tracerStart) then {
-			_finalPosition = [];
-		} else {
-			_finalPosition = _lastTracerPosition;
-		};
+		_finalPosition = _lastValidTracerPosition;
 	};
 
-	//If we reach the destination, we can spawn at the target location without an issue
+	// @dijksterhuis: If we are very close to the end of the projected line
+	// then we need to start spawning in AI at the last known good position.
+	// this means AI spawn one line step further away from the objective,
+	// but this is necessary to avoid AI units spawning at water positions
+	// when setting _finalPosition = _tracerPosition
 	if (_tracerPosition distance2D _tracerEndPos < _stepSize) exitWith {
-		_finalPosition = _tracerEndPos;
+		_finalPosition = _tracerPosition;
 	};
 };
 
 if (!isNil "debugAttackTracer") then {
-	private _mark = createMarker ["TracerFinal" + str time, _tracerPosition];
+	private _mark = createMarker ["TracerFinal" + str time, _finalPosition];
 	_mark setMarkerType "mil_dot";
 	_mark setMarkerColor "ColorRed";
 	tracerMarkers pushBack _mark;
 };
 
+if (_finalPosition isEqualTo _tracerStart) then {
+	_finalPosition = [];
```

Comment:
Note to self -- add a comment about what happens when we return an empty array as final position.